components/camel-pulsar/src/main/java/org.apache.camel.component.pulsar/configuration/PulsarEndpointConfiguration.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.pulsar.configuration;
+
+import org.apache.camel.component.pulsar.utils.consumers.SubscriptionType;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriParams;
+
+import static org.apache.camel.component.pulsar.utils.consumers.SubscriptionType.EXCLUSIVE;
+
+@UriParams
+public class PulsarEndpointConfiguration {
+
+    @UriParam(label = "consumer", description = "Name of the subscription to use", defaultValue = "subscription")
+    private String subscriptionName = "subs";
+    @UriParam(label = "consumer", description = "Type of the subscription", enums = "EXCLUSIVE, SHARED, FAILOVER", defaultValue = "EXCLUSIVE")
```

Comment:
you can take as example any other configuration, we usually add the description as javadoc, then this information will be picked up while generating the autodocumentation